File: examples/purchase-tester-java/src/main/java/com/revenuecat/purchasetester/java/CatsActivity.java
package com.revenuecat.purchasetester.java;

import static com.revenuecat.purchasetester.java.MainApplication.PREMIUM_ENTITLEMENT_ID;

import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.revenuecat.purchases.CustomerInfo;
import com.revenuecat.purchases.EntitlementInfo;
import com.revenuecat.purchases.Purchases;
import com.revenuecat.purchases.PurchasesError;
import com.revenuecat.purchases.interfaces.ReceiveCustomerInfoCallback;

import java.text.DateFormat;
import java.util.Date;

public class CatsActivity extends AppCompatActivity {
    Button restoreView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_cats);

        findViewById(R.id.go_premium).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });

        restoreView = findViewById(R.id.purchase_restore);
        restoreView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                restoreView.setText("Loading...");
                Purchases.getSharedInstance().restorePurchases(new ReceiveCustomerInfoCallback() {
                    @Override
                    public void onReceived(@NonNull CustomerInfo customerInfo) {
                        restoreView.setText("Restore Purchases");
                        configureContent(customerInfo);
                    }

                    @Override
                    public void onError(@NonNull PurchasesError error) {
                        Log.e("Purchase Tester Java", error.getMessage());
                    }
                });
            }
        });
    }

    private void configureContent(final CustomerInfo customerInfo) {
        TextView purchaseDateView = findViewById(R.id.purchase_date_label);
        TextView expirationDateView = findViewById(R.id.expiration_date_label);
        TextView catIconView = findViewById(R.id.cat_content_label);
        View goPremiumView = findViewById(R.id.go_premium);

        EntitlementInfo proEntitlement = customerInfo.getEntitlements().get(PREMIUM_ENTITLEMENT_ID);
        if (proEntitlement != null && proEntitlement.isActive()) {
            Log.i("Purchase Tester Java", "Hey there premium, you're a happy cat 😻");
            catIconView.setText("😻");
            goPremiumView.setVisibility(View.GONE);
            restoreView.setVisibility(View.GONE);

            DateFormat dateFormat = android.text.format.DateFormat.getDateFormat(getApplicationContext());

            Date purchaseDate = customerInfo.getPurchaseDateForEntitlement(PREMIUM_ENTITLEMENT_ID);
            purchaseDateView.setText("Purchase Date: " + dateFormat.format(purchaseDate));
            purchaseDateView.setVisibility(View.VISIBLE);

            Date expirationDate = customerInfo.getExpirationDateForEntitlement(PREMIUM_ENTITLEMENT_ID);
            expirationDateView.setText("Expiration Date: " + dateFormat.format(expirationDate));
            expirationDateView.setVisibility(View.VISIBLE);
        } else {
            Log.i("Purchase Tester Java", "Happy cats are only for premium members 😿");
            catIconView.setText("😿");
            goPremiumView.setVisibility(View.VISIBLE);
            restoreView.setVisibility(View.VISIBLE);
            expirationDateView.setVisibility(View.GONE);
            purchaseDateView.setVisibility(View.GONE);
        }

        Button manageSubscriptionButton = findViewById(R.id.manage_subscription);

        if (customerInfo.getManagementURL() != null) {
            manageSubscriptionButton.setVisibility(View.VISIBLE);
            manageSubscriptionButton.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    try {
                        startActivity(new Intent(Intent.ACTION_VIEW, customerInfo.getManagementURL()));
                    } catch (ActivityNotFoundException exception) {
                        exception.printStackTrace();
                    }
                }
            });
        } else {
            manageSubscriptionButton.setVisibility(View.GONE);
        }
    }

}

File: purchases-sample/src/main/java/com/revenuecat/sample/CatsActivity.kt
package com.revenuecat.sample

import android.content.ActivityNotFoundException
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.util.Log
import android.view.View.GONE
import android.view.View.VISIBLE
import androidx.appcompat.app.AppCompatActivity
import com.revenuecat.purchases.PurchaserInfo
import com.revenuecat.purchases.Purchases
import com.revenuecat.purchases.getPurchaserInfoWith
import com.revenuecat.purchases.restorePurchasesWith
import kotlinx.android.synthetic.main.activity_cats.*

class CatsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_cats)

        go_premium.setOnClickListener {
            finish()
        }
        purchase_restore.setOnClickListener {
            purchase_restore.text = "Loading..."
            Purchases.sharedInstance.restorePurchasesWith(::showError) { purchaserInfo ->
                purchase_restore.text = "Restore Purchases"
                configureContent(purchaserInfo)
            }
        }
    }

    override fun onResume() {
        super.onResume()
        Purchases.sharedInstance.getPurchaserInfoWith(::showError, ::configureContent)
    }

    private fun configureContent(purchaserInfo: PurchaserInfo) {
        if (purchaserInfo.entitlements[PREMIUM_ENTITLEMENT_ID]?.isActive == true) {
            Log.i("Purchases Sample", "Hey there premium, you're a happy cat 😻")
            cat_content_label.text = "😻"
            go_premium.visibility = GONE
            purchase_restore.visibility = GONE
            val dateFormat = android.text.format.DateFormat.getDateFormat(applicationContext)

            purchaserInfo.getPurchaseDateForEntitlement(PREMIUM_ENTITLEMENT_ID)?.let {
                purchase_date_label.text = "Purchase Date: ${dateFormat.format(it)}"
                purchase_date_label.visibility = VISIBLE
            }

            purchaserInfo.getExpirationDateForEntitlement(PREMIUM_ENTITLEMENT_ID)?.let {
                expiration_date_label.text = "Expiration Date: ${dateFormat.format(it)}"
                expiration_date_label.visibility = VISIBLE
            }
        } else {
            Log.i("Purchases Sample", "Happy cats are only for premium members \uD83D\uDE3F")
            cat_content_label.text = "😿"
            go_premium.visibility = VISIBLE
            purchase_restore.visibility = VISIBLE
            expiration_date_label.visibility = GONE
            purchase_date_label.visibility = GONE
        }

        if (purchaserInfo.managementURL != null) {
            manage_subscription.visibility = VISIBLE
            manage_subscription.setOnClickListener {
                try {
                    startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(purchaserInfo.managementURL)))
                } catch (e: ActivityNotFoundException) {
                    e.printStackTrace()
                }
            }
        } else {
            manage_subscription.visibility = GONE
        }
    }
}
